src/test/scala/chiselTests/reflect/DataMirrorSpec.scala

```diff
@@ -266,6 +266,22 @@ class DataMirrorSpec extends ChiselFlatSpec {
     })
   }
 
+  "currentInstancePorts" should "return an Instance's IOs" in {
+    @instantiable
+    class Foo extends RawModule {
+      @public val in = IO(Input(Bool()))
+      @public val out = IO(Output(Bool()))
+    }
+
+    ChiselStage.emitCHIRRTL(new RawModule {
+      val foo = Instantiate(new Foo)
+
+      val ports = DataMirror.internal.currentInstancePorts(foo)
+
+      ports should be(Seq(foo.in, foo.out))
```

Comment:
You should also test a `Module(new Foo)`, if possible, also see if you can box the anonymous `RawModule` in an `Instance` and check its ports (if so, give it a port).

Reply:
Thanks for the pointers, added in [6b378ea](https://github.com/chipsalliance/chisel/pull/4362/commits/6b378ea9f8f638b8b8b09e671b21763354869f3f)